src/vs/workbench/contrib/issue/electron-sandbox/issueFormService.ts

```diff
@@ -0,0 +1,212 @@
+import { IIssueFormService, IssueReporterData, IssueReporterWindowConfiguration } from 'vs/workbench/contrib/issue/common/issue';
+import { IssueReporter2 } from 'vs/workbench/contrib/issue/electron-sandbox/issueReporterService2';
+import { AuxiliaryWindowMode, IAuxiliaryWindowService } from 'vs/workbench/services/auxiliaryWindow/browser/auxiliaryWindowService';
+
+export class IssueFormService2 implements IIssueFormService {
+
+	private configuration: IssueReporterWindowConfiguration | undefined;
+
+	private extensionIdentifierSet: ExtensionIdentifierSet = new ExtensionIdentifierSet();
+
+	declare readonly _serviceBrand: undefined;
+
+	private currentData: IssueReporterData | undefined;
+
+	private issueReporterWindow: Window | null = null;
+	private issueReporterParentWindow: BrowserWindow | null = null;
+
+	constructor(
+		@IInstantiationService private readonly instantiationService: IInstantiationService,
+		@INativeEnvironmentService private readonly environmentService: INativeEnvironmentService,
+		@ILogService private readonly logService: ILogService,
+		@IDialogService private readonly dialogService: IDialogService,
+		@INativeHostService private readonly nativeHostService: INativeHostService,
+		@IAuxiliaryWindowService private readonly auxiliaryWindowService: IAuxiliaryWindowService,
+		@IMenuService private readonly menuService: IMenuService,
+		@IContextKeyService private readonly contextKeyService: IContextKeyService,
+	) { }
+
+	//#region Used by renderer
+	async openReporter(data: IssueReporterData): Promise<void> {
+		if (data.extensionId && this.extensionIdentifierSet.has(data.extensionId)) {
+			this.currentData = data;
+			this.issueReporterWindow?.focus();
+			return;
+		}
+
+		if (this.issueReporterWindow) {
+			this.issueReporterWindow.focus();
+			return;
+		}
+		this.openAuxIssueReporter(data);
+	}
+
+	async openAuxIssueReporter(data: IssueReporterData): Promise<void> {
+		const disposables = new DisposableStore();
+
+		// Auxiliary Window
+		const auxiliaryWindow = disposables.add(await this.auxiliaryWindowService.open({ mode: AuxiliaryWindowMode.Custom, bounds: { width: 700, height: 800 } }));
+		this.issueReporterWindow = auxiliaryWindow.window;
+
+		if (auxiliaryWindow) {
+			await auxiliaryWindow.whenStylesHaveLoaded;
+			auxiliaryWindow.window.document.title = 'Issue Reporter';
+			auxiliaryWindow.window.document.body.classList.add('issue-reporter-body');
+
+			// custom issue reporter wrapper
+			const div = document.createElement('div');
+			div.classList.add('monaco-workbench');
+
+			// removes preset monaco-workbench
+			auxiliaryWindow.container.remove();
+			auxiliaryWindow.window.document.body.appendChild(div);
+			safeInnerHtml(div, BaseHtml());
+
+			// Store into config object URL
+			this.configuration = {
+				appRoot: this.environmentService.appRoot,
+				windowId: 0,
+				userEnv: {},
+				data,
+				disableExtensions: !!this.environmentService.disableExtensions,
+				os: {
+					type: '',
```

Comment:
not sure if I have a good replacement for this. I tried looking for things in global, platform, etc, not sure if there's anything that is exactly the same as `{ arch, release, type } from 'os';`

Reply:
Don't ask "how do I get these in a web environment" and instead ask "what would be useful to capture in a web environment".

For the web side, we probably want to know:
* The OS
* The browser

[platform.ts](https://github.com/microsoft/vscode/blob/ce4bf35469ca88ab5bf1c86aca1bff385c5b5e84/src/vs/base/common/platform.ts#L1) has both of these friend.

For desktop, we do still want these 3 things.

Reply:
fixed using nativeHostService for desktop!